### DISCUSSION TITLE
Multiple explainers

### DISCUSSION THREAD

**Discussion by farridav** [oegedijk/explainerdashboard#306] [Q&A]
I have a pipeline in which we create a shap explainer for each group of the data (as it offers more localised insight).

Is there a way for me to choose the explainer in which to render the dashboard, within the dashboard? 

I figure I could make this work with inline dashboards, I'm just wondering if this is the right way .

Thanks for the hard work, such a brilliant project

**Comment by oegedijk** [comment]
hi, it should be possible, but will require a little bit of hacking.

So if you look here where the shap_explainer property is defined you can see it dynamically creates a `self._shap_explainer` attribute whenever first accessed. 

https://github.com/oegedijk/explainerdashboard/blob/78cb10f63223604f132fae7247643b1468aa116b/explainerdashboard/explainers.py#L1053

So if you just set this attribute immediately after initalizing the explainer that it should just use your TreeExplainer. 

Not sure what the purpose of this is though? Initializing the explainer is not the intensive part, its generating the shap values. (and these can also be inserted in the explainer if you already have them calculated btw)
